dependency_cache_test.go

```diff
@@ -317,6 +319,21 @@ func testDependencyCache(t *testing.T, context spec.G, it spec.S) {
 			Expect(io.ReadAll(a)).To(Equal([]byte("alternate-fixture")))
 		})
 
+		it("sets downloaded file name to uri's sha256 with empty SHA256 and query parameters in the uri", func() {
+			dependency.SHA256 = ""
+			dependency.URI = fmt.Sprintf("%s/test-path?param1=value1&param2=value2", server.URL())
+			server.AppendHandlers(ghttp.RespondWith(http.StatusOK, "alternate-fixture"))
+
+			hasher := sha256.New()
+			hasher.Write([]byte(dependency.URI))
+
+			a, err := dependencyCache.Artifact(dependency)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(io.ReadAll(a)).To(Equal([]byte("alternate-fixture")))
+			Expect(filepath.Base(a.Name())).To(Equal(hex.EncodeToString(hasher.Sum(nil))))
```

Comment:
instead of re doing what you do in the implementation (`hasher` creation), you could have just compared the test filename with the String representation of the hash?

Reply:
yeah, this is what I usually do; since the shasum is not gonna change from one run to another, then instead of repeating what you did in the implementation, you could just write your test with this assumption:  

> I know the shasum of `http://example.com/test-path?param1=value1&param2=value2` is `ffff` so I assert the impl. will return `ffff` as the filename

That way, you decouple your knowledge of the underlying code, not caring how the impl. code was written.

I hope that makes sense

Reply:
Ah ok, it makes sense, but when I tried comparing directly with the shasum, but it seems the `httptest` library is [randomly selecting ](https://cs.opensource.google/go/go/+/refs/tags/go1.21.0:src/net/http/httptest/server.go;l=24)an available port for the mocked server and that in turn keeps changing the dependency uri, so I couldn't find a way to hardcode the shasum in the comparison.

Reply:
ah! that's a good point, the URL is dynamic, I had not noticed.

hum, well I guess you can not predict the shasum then 🤷

In that case, I believe you already chose for your test the best outcome!

All good!

Reply:
haha, purely accidently
hadn't noticed only until now that the url is not static